src/unix/solarish/mod.rs

```diff
@@ -2182,11 +2232,22 @@ pub const PORT_SOURCE_MQ: ::c_int = 6;
 pub const PORT_SOURCE_FILE: ::c_int = 7;
 
 pub const NONROOT_USR: ::c_short = 2;
+cfg_if! {
+    if #[cfg(target_os = "illumos")] {
 pub const _UTX_USERSIZE: usize = 32;
 pub const _UTX_LINESIZE: usize = 32;
 pub const _UTX_PADSIZE: usize = 5;
 pub const _UTX_IDSIZE: usize = 4;
 pub const _UTX_HOSTSIZE: usize = 257;
+    }
+}
+cfg_if! {
+    if #[cfg(target_os = "solaris")] {
+pub const _UTMP_USER_LEN: usize = 32;
+pub const _UTMP_LINE_LEN: usize = 32;
+pub const _UTMP_ID_LEN: usize = 4;
+    }
+}
```

Comment:
Here and elsewhere: Move to the OS specific file, rather than a bunch of cfg_if stuff? (At least when it's huge blocks of stuff, vs one-off things in a largely shared list)